components/connector/internal/revocation/loader.go

```diff
@@ -0,0 +1,109 @@
+package revocation
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type Loader interface {
+	Run(ctx context.Context)
+}
+
+type revocationListLoader struct {
+	revocationListCache Cache
+	manager             Manager
+	configMapName       string
+	reconnectInternal   time.Duration
+}
+
+func NewRevocationListLoader(revocationListCache Cache,
+	manager Manager,
+	configMapName string,
+	reconnectInternal time.Duration,
+) Loader {
+	return &revocationListLoader{
+		revocationListCache: revocationListCache,
+		manager:             manager,
+		configMapName:       configMapName,
+		reconnectInternal:   reconnectInternal,
+	}
+}
+
+func (rl *revocationListLoader) Run(ctx context.Context) {
+	rl.startKubeWatch(ctx)
+}
+
+func (rl *revocationListLoader) startKubeWatch(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			log.Println("context cancelled, stop revocation config map watcher")
+			return
+		default:
+		}
+		watcher, err := connectWatch(rl.manager, rl.configMapName)
+		if err != nil {
+			log.Printf("Could not initialize watcher: %s. Sleep for %s and try again...", err, rl.reconnectInternal.String())
+			time.Sleep(rl.reconnectInternal)
+			continue
+		}
+		log.Println("Waiting for revocation list configmap events...")
+		for {
```

Comment:
Why is this for loop needed since there is a outside loop and we can processEvents has loop also. Here in the if bread can be continue and it will bi sufficient?

Reply:
overlooked one of cases in which we don't need reconnect, but yes you are right, it can be handleded in the processEvents